website/docs/cloud-docs/api-docs/audit-trails.mdx

```diff
@@ -108,23 +108,334 @@ $ curl \
+
+The following sections list the audit log events containing both the standard response schema and a specific payload for each action.
+
+### Data retention policy events
+
+You can define [data retention policies](/terraform/cloud-docs/workspaces/settings/deletion#data-retention-policies) to help reduce object storage consumption. 
+
+#### Destroy
+
+A Terraform Cloud organization emits this event when it destroys a data retention policy(`data_retention_policy`). Alongside the [standard audit trail fields](#standard-response-fields), this event response includes the following fields:
+
+| Key                        | Data Type | Description                                             |
+| ---                        | ---       | ---                                                     |
+| `delete_older_than_n_days` | integer   | The number of days to retain data before deletion. Must be greater than 0. |
+| `organization`             | string    | The organization associated with the data retention policy. |
+| `organization_id`          | string    | The ID of the organization associated with the data retention policy. |
+
+### Project events
+
+[Projects](/terraform/cloud-docs/workspaces/organize-workspaces-with-projects) let you organize your workspaces and scope access to workspace resources.
+
+#### Add Team
+
+A Terraform Cloud organization emits this event when someone in your organization adds a team(`add_team`) to a project. Alongside the [standard audit trail fields](#standard-response-fields), this event response includes the following fields:
+
+| Key          | Data Type | Description                           |
+| ---          | ---       | ---                                   |
+| `team`       | string    | The name or identifier of the team.    |
+| `permissions` | string   | The permissions granted to the team.  |
+
+#### Update Team Permissions
+
+A Terraform Cloud organization emits this event when someone in your organization updates the permissions(`update_team_permissions`) of a team on a project. Alongside the [standard audit trail fields](#standard-response-fields), this event response includes the following fields:
+
+| Key          | Data Type | Description                           |
+| ---          | ---       | ---                                   |
+| `team`       | string    | The name or identifier of the team.    |
+| `permissions` | string   | The updated permissions for the team. |
+
+#### Remove Team
+
+A Terraform Cloud organization emits this event when someone in your organization removes a team(`remove_team`) from a project. Alongside the [standard audit trail fields](#standard-response-fields), this event response includes the following fields:
+
+| Key          | Data Type | Description                           |
+| ---          | ---       | ---                                   |
+| `team`       | string    | The name or identifier of the team.    |
+
+### Policy checks
```

Comment:
@carolinaborim I did my best representing policy checks and policy evaluations but, I honestly found this section of the spreadsheet a little confusing, so please double check this for accuracy and suggest or adjust as you see fit!

Reply:
@dylanegan can you help double check the policy evaluations?

Reply:
Oh now I see what is happening here. The mapping in the spreadsheet is still too code heavy. So for each `process_audit_log_event`, it is actually happening for a discrete action, which is when the policy check or evaluation is processing through the state machine.

```
process_audit_log_event(action: :queued)
process_audit_log_event(action: :passed, include_result: true)
process_audit_log_event(action: :soft_failed, include_result: true)
process_audit_log_event(action: :hard_failed, include_result: true)
process_audit_log_event(action: :overridden, actor: user, comment: comment)
process_audit_log_event(action: :canceled, actor: user, comment: comment)
process_audit_log_event(action: :force_canceled, actor: user, comment: comment)
process_audit_log_event(action: :errored, include_result: true)
```

This would equate to different actions being logged, i.e. queued, passed, soft_failed, etc.

Each action is logged with the default payload of

```
{
  comment: comment (default null),
  run: {
    id: self.run.external_id,
    message: self.run.message,
  },
  workspace: {
    id: self.run.workspace.external_id,
    name: self.run.workspace.name,
    },
}
```

Where `comment` is supplied, it'll be a string I believe, although @carolinaborim can check that against an API query.

Where `include_result` is included the payload is expanded upon. If it has a result... 😅 

```
{
  result: self.result.result,
  passed: self.result.passed,
  total_failed: self.result.total_failed,
  hard_failed: self.result.hard_failed,
  soft_failed: self.result.soft_failed,
  advisory_failed: self.result.advisory_failed,
  duration_ms: self.result.duration_ms,
  sentinel: self.result.sentinel,
}
```

@carolinaborim you'd need to look up the types for these, but I assume most are numeric, with some being boolean and a version, perhaps?

This also needs to be reassessed for evaluations, which is slightly different.

```
process_audit_log_event(action: :queue, actor: user, include_result: false, comment: comment)
--
process_audit_log_event(action: :run, actor: user, include_result: false, comment: comment)
process_audit_log_event(action: :pass, actor: user, include_result: true, comment: comment)
process_audit_log_event(action: :fail, actor: user, include_result: true, comment: comment)
process_audit_log_event(action: :override, actor: user, include_result: false, comment: comment)
process_audit_log_event(action: :error, actor: user, include_result: true, comment: comment)
process_audit_log_event(action: :cancel, actor: user, include_result: false, comment: comment)
process_audit_log_event(action: :force_cancel, actor: user, include_result: false, comment: comment)
```

@carolinaborim something similar to what I posted should help @rkoron007 out.

Reply:
Ok. I will just update the spreadsheet with the types because I think it would be easier for @rkoron007 to follow her flow from there!

Reply:
So, the are two things... the Column C on the spreadsheet is related to the first piece os Dylan's comment and the payload of the action is actually already described (with types) on the first part of the Column D Line 234:
![image](https://github.com/hashicorp/terraform-docs-common/assets/7681028/dbac3b51-519b-4e69-813e-142d2984e2ab)

but basically: 
`comment: string,
  run: {
    id: string,
    message: string,
  },
  workspace: {
    id: string,
    name: string,
    },
  }`

Reply:
Last comment: for  `policy_evaluation`
Column C is the same idea is the first piece of Dylan's comment - the actions are being logged with a payload that contains a comment (string).

In this case, the payload can contain different information related to Workspace and related to the Run, but the info is on Column D, line 261:
![image](https://github.com/hashicorp/terraform-docs-common/assets/7681028/65e7b1e4-08b8-4bb3-aa69-b45a139c144c)

 comment: string

  payload[workspace] = {
    id: string, (workspace ID)
    name: string, (workspace name)
  }
  payload[run] = {
    id: string, (run ID)
    message: string, (policy message),
    actor: string
  }
  actor: string (policy owner)

Reply:
Thank you @dylanegan and @carolinaborim!

How does `include_result` affect policy evaluations?  I can see clearly what Policy Checks return everytime, but I don't know what Policy evaluations return everytime v.s. what they return when a result is included.